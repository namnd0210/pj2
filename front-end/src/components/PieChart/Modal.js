import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import _ from 'lodash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck, faAngleDown } from '@fortawesome/free-solid-svg-icons';

import { fetchPieChartData } from '../../actions';

export default ({ startDate, endDate }) => {
  const { data } = useSelector(state => state.pieChartData)
  const [handleData, setHandleData] = useState(data);
  const dispatch = useDispatch();
  const [modal, setModal] = useState(false);

  const toggle = () => {
    setModal(!modal)
    if(modal) dispatch(fetchPieChartData(startDate, endDate, handleData))
  };

  const onClick = (item, index) => {
    setHandleData([
      ..._.slice(handleData, 0, index),
      { ...item, isActive: !item.isActive },
      ..._.slice(handleData, index + 1)
    ])
  }

  return (
    <div>
      <Button color="success" onClick={toggle} style={{ padding: "10px 20px" }}>+</Button>
      <Modal isOpen={modal} toggle={toggle}>
        <ModalHeader toggle={toggle} charCode={<FontAwesomeIcon icon={faAngleDown} />}></ModalHeader>
        <ModalBody>
          <div>
            {
              _.map(handleData, (item, index) =>
                <div
                  className="row"
                  onClick={() => onClick(item, index)}
                  style={{ cursor: "pointer" }}
                >
                  <div className="col-10">{item.x}</div>
                  {item.isActive &&
                    <div className="col-2">
                      <FontAwesomeIcon icon={faCheck} />
                    </div>
                  }
                </div>
              )
            }
          </div>
        </ModalBody>
        <ModalFooter>
          <Button color="primary" onClick={toggle}>OK</Button>
        </ModalFooter>
      </Modal>
    </div>
  );
}
