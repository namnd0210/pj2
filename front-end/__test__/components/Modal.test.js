import React from 'react'
import Modal from '../../src/components/Analysis/PieChart/Modal'
import rerender from 'react-test-renderer'

jest.mock('react-redux', () => ({
  useDispatch: () => { },
  useSelector: () => ({})
}));

test('Clicked Modal', () => {
  const modal = rerender.create(
    <Modal startDate='' endDate='' />
  )
  let tree = modal.toJSON();
  expect(tree).toMatchSnapshot();
})