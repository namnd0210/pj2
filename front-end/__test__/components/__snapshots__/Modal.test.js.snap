// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicked Modal 1`] = `
<div>
  <button
    aria-label={null}
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "padding": "10px 20px",
      }
    }
    type="button"
  >
    +
  </button>
</div>
`;
